pandas/core/internals.py

```diff
@@ -1529,6 +1529,18 @@ def to_native_types(self, slicer=None, na_rep='', float_format=None,
         if slicer is not None:
             values = values[:, slicer]
 
+        # no special formatting is desired
+        if float_format is None and decimal == '.':
```

Comment:
I guess quoting *only* matters in a csv (and not generally to output, which is where the Formatters matter). pls add a comment with the issue number